### ISSUE TITLE
An unknown error when using yolov3_darknet53 to train the fruit dataset

### ISSUE THREAD

**Issue by p1n0cch10** [PaddlePaddle/PaddleDetection#527]
I am using yolov3 darknet53 to train the fruit dataset, but an error occurs.

The command is
`python -u tools/train.py -c configs/yolov3_darknet_myfruit.yml -o use_fine_grained_loss=true --eval`

The output is something like
`2020-04-17 19:10:48,915-INFO: Downloading DarkNet53_pretrained.tar from https://paddle-imagenet-models-name.bj.bcebos.com/DarkNet53_pretrained.tar
100%|████████████████████████████████| 158940/158940 [00:09<00:00, 16988.39KB/s]
2020-04-17 19:10:58,483-INFO: Decompressing /home/alex/.cache/paddle/weights/DarkNet53_pretrained.tar...
2020-04-17 19:10:58,643-INFO: Loading parameters from /home/alex/.cache/paddle/weights/DarkNet53_pretrained...
2020-04-17 19:10:58,772-WARNING: /tmp/tmp9gh3g4dv/DarkNet53_pretrained.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
2020-04-17 19:10:58,772-WARNING: /tmp/tmp9gh3g4dv/DarkNet53_pretrained.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
2020-04-17 19:10:59,074-INFO: 240 samples in file dataset/fruit/train.txt
2020-04-17 19:11:05,588-INFO: places would be ommited when DataLoader is not iterable
I0417 19:11:05.609894 30521 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 1 cards are used, so 1 programs are executed in parallel.
I0417 19:11:05.657047 30521 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1
I0417 19:11:05.815907 30521 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True
I0417 19:11:05.851222 30521 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0
2020-04-17 19:11:06,004-WARNING: recv endsignal from outq with errmsg[consumer[consumer-0ee-1] failed to map with error:[target0 not in samples]]
`

The yolov3_darknet_myfruit.yml is modified according to yolov3_mobilenet_v1_fruit.yml.
I just change .yml to .txt inorder to attach the file to a github post.
[yolov3_darknet_myfruit.txt](https://github.com/PaddlePaddle/PaddleDetection/files/4492587/yolov3_darknet_myfruit.txt)

Many thanks for helping me.

**Comment by p1n0cch10** [comment]
I update my .yml a little bitter but another error occurs.
[yolov3_darknet_fruit_test.txt](https://github.com/PaddlePaddle/PaddleDetection/files/4493741/yolov3_darknet_fruit_test.txt)

When I execute

`python -u tools/train.py -c configs/yolov3_darknet_fruit_test.yml -o use_fine_grained_loss=true --eval`

it shows

`Traceback (most recent call last):
  File "tools/train.py", line 334, in <module>
    main()
  File "tools/train.py", line 143, in main
    eval_reader = create_reader(cfg.EvalReader, devices_num=1)
  File "/home/alex/Documents/tutorial/paddle/PaddleDetection/ppdet/data/reader.py", line 410, in create_reader
    reader = Reader(**cfg)()
  File "/home/alex/Documents/tutorial/paddle/PaddleDetection/ppdet/data/reader.py", line 204, in __init__
    self._roidbs = self._dataset.get_roidb()
  File "/home/alex/Documents/tutorial/paddle/PaddleDetection/ppdet/data/source/dataset.py", line 68, in get_roidb
    self.load_roidb_and_cname2cid()
  File "/home/alex/Documents/tutorial/paddle/PaddleDetection/ppdet/data/source/voc.py", line 131, in load_roidb_and_cname2cid
    gt_class[i][0] = cname2cid[cname]
KeyError: 'orange'
`



when I execute

`python -u tools/train.py -c configs/yolov3_darknet_fruit_test.yml -o use_fine_grained_loss=true`

it shows

`/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
Traceback (most recent call last):
  File "tools/train.py", line 334, in <module>
    main()
  File "tools/train.py", line 175, in main
    exe.run(startup_prog)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 783, in run
    six.reraise(*sys.exc_info())
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 778, in run
    use_program_cache=use_program_cache)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 831, in _run_impl
    use_program_cache=use_program_cache)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 905, in _run_program
    fetch_var_name)
paddle.fluid.core_avx.EnforceNotMet: 


C++ Call Stacks (More useful to developers):

0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::platform::CUDADeviceContext::CUDADeviceContext(paddle::platform::CUDAPlace)
3   std::_Function_handler<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > (), std::reference_wrapper<std::_Bind_simple<paddle::platform::EmplaceDeviceContext<paddle::platform::CUDADeviceContext, paddle::platform::CUDAPlace>(std::map<paddle::platform::Place, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::less<paddle::platform::Place>, std::allocator<std::pair<paddle::platform::Place const, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > > > >*, paddle::platform::Place)::{lambda()#1} ()> > >::_M_invoke(std::_Any_data const&)
4   std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::__future_base::_Result_base::_Deleter>, std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > >::_M_invoke(std::_Any_data const&)
5   std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&)
6   std::__future_base::_Deferred_state<std::_Bind_simple<paddle::platform::EmplaceDeviceContext<paddle::platform::CUDADeviceContext, paddle::platform::CUDAPlace>(std::map<paddle::platform::Place, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::less<paddle::platform::Place>, std::allocator<std::pair<paddle::platform::Place const, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > > > >*, paddle::platform::Place)::{lambda()#1} ()>, std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >::_M_run_deferred()
7   paddle::platform::DeviceContextPool::Get(paddle::platform::Place const&)
8   paddle::framework::GarbageCollector::GarbageCollector(paddle::platform::Place const&, unsigned long)
9   paddle::framework::UnsafeFastGPUGarbageCollector::UnsafeFastGPUGarbageCollector(paddle::platform::CUDAPlace const&, unsigned long)
10  paddle::framework::Executor::RunPreparedContext(paddle::framework::ExecutorPrepareContext*, paddle::framework::Scope*, bool, bool, bool)
11  paddle::framework::Executor::Run(paddle::framework::ProgramDesc const&, paddle::framework::Scope*, int, bool, bool, std::vector<std::string, std::allocator<std::string> > const&, bool, bool)

----------------------
Error Message Summary:
----------------------
Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information: out of memory at (/paddle/paddle/fluid/platform/device_context.cc:221)`

**Comment by p1n0cch10** [comment]
I update my .yml file a little bit according to yolov3_mobilenet_v1_fruit.yml
[yolov3_darknet_fruit_test.txt](https://github.com/PaddlePaddle/PaddleDetection/files/4493810/yolov3_darknet_fruit_test.txt)

After I execute

`python -u tools/train.py -c configs/yolov3_darknet_fruit_test.yml -o use_fine_grained_loss=true --eval`

an unknown error occurs. Could anyone help me with the error. Many thanks.


`2020-04-17 23:32:33,731-INFO: 60 samples in file dataset/fruit/val.txt
2020-04-17 23:32:33,732-INFO: places would be ommited when DataLoader is not iterable
/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
Traceback (most recent call last):
  File "tools/train.py", line 331, in <module>
    main()
  File "tools/train.py", line 172, in main
    exe.run(startup_prog)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 783, in run
    six.reraise(*sys.exc_info())
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 778, in run
    use_program_cache=use_program_cache)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 831, in _run_impl
    use_program_cache=use_program_cache)
  File "/home/alex/paddle-venv/lib/python3.6/site-packages/paddle/fluid/executor.py", line 905, in _run_program
    fetch_var_name)
paddle.fluid.core_avx.EnforceNotMet: 


C++ Call Stacks (More useful to developers):

0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::platform::CUDADeviceContext::CUDADeviceContext(paddle::platform::CUDAPlace)
3   std::_Function_handler<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > (), std::reference_wrapper<std::_Bind_simple<paddle::platform::EmplaceDeviceContext<paddle::platform::CUDADeviceContext, paddle::platform::CUDAPlace>(std::map<paddle::platform::Place, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::less<paddle::platform::Place>, std::allocator<std::pair<paddle::platform::Place const, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > > > >*, paddle::platform::Place)::{lambda()#1} ()> > >::_M_invoke(std::_Any_data const&)
4   std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::__future_base::_Result_base::_Deleter>, std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > >::_M_invoke(std::_Any_data const&)
5   std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&)
6   std::__future_base::_Deferred_state<std::_Bind_simple<paddle::platform::EmplaceDeviceContext<paddle::platform::CUDADeviceContext, paddle::platform::CUDAPlace>(std::map<paddle::platform::Place, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >, std::less<paddle::platform::Place>, std::allocator<std::pair<paddle::platform::Place const, std::shared_future<std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > > > > >*, paddle::platform::Place)::{lambda()#1} ()>, std::unique_ptr<paddle::platform::DeviceContext, std::default_delete<paddle::platform::DeviceContext> > >::_M_run_deferred()
7   paddle::platform::DeviceContextPool::Get(paddle::platform::Place const&)
8   paddle::framework::GarbageCollector::GarbageCollector(paddle::platform::Place const&, unsigned long)
9   paddle::framework::UnsafeFastGPUGarbageCollector::UnsafeFastGPUGarbageCollector(paddle::platform::CUDAPlace const&, unsigned long)
10  paddle::framework::Executor::RunPreparedContext(paddle::framework::ExecutorPrepareContext*, paddle::framework::Scope*, bool, bool, bool)
11  paddle::framework::Executor::Run(paddle::framework::ProgramDesc const&, paddle::framework::Scope*, int, bool, bool, std::vector<std::string, std::allocator<std::string> > const&, bool, bool)


Error Message Summary:

Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information: out of memory at (/paddle/paddle/fluid/platform/device_context.cc:221)`

**Comment by p1n0cch10** [comment]
> I am using yolov3 darknet53 to train the fruit dataset, but an error occurs.
> 
> The command is
> `python -u tools/train.py -c configs/yolov3_darknet_myfruit.yml -o use_fine_grained_loss=true --eval`
> 
> The output is something like
> `2020-04-17 19:10:48,915-INFO: Downloading DarkNet53_pretrained.tar from https://paddle-imagenet-models-name.bj.bcebos.com/DarkNet53_pretrained.tar 100%|████████████████████████████████| 158940/158940 [00:09<00:00, 16988.39KB/s] 2020-04-17 19:10:58,483-INFO: Decompressing /home/alex/.cache/paddle/weights/DarkNet53_pretrained.tar... 2020-04-17 19:10:58,643-INFO: Loading parameters from /home/alex/.cache/paddle/weights/DarkNet53_pretrained... 2020-04-17 19:10:58,772-WARNING: /tmp/tmp9gh3g4dv/DarkNet53_pretrained.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ] 2020-04-17 19:10:58,772-WARNING: /tmp/tmp9gh3g4dv/DarkNet53_pretrained.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ] 2020-04-17 19:10:59,074-INFO: 240 samples in file dataset/fruit/train.txt 2020-04-17 19:11:05,588-INFO: places would be ommited when DataLoader is not iterable I0417 19:11:05.609894 30521 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 1 cards are used, so 1 programs are executed in parallel. I0417 19:11:05.657047 30521 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1 I0417 19:11:05.815907 30521 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True I0417 19:11:05.851222 30521 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0 2020-04-17 19:11:06,004-WARNING: recv endsignal from outq with errmsg[consumer[consumer-0ee-1] failed to map with error:[target0 not in samples]] `
> 
> The yolov3_darknet_myfruit.yml is modified according to yolov3_mobilenet_v1_fruit.yml.
> I just change .yml to .txt inorder to attach the file to a github post.
> [yolov3_darknet_myfruit.txt](https://github.com/PaddlePaddle/PaddleDetection/files/4492587/yolov3_darknet_myfruit.txt)
> 
> Many thanks for helping me.
